### DISCUSSION TITLE
Filter fields in .udf file

### DISCUSSION THREAD

**Discussion by aiskhak** [Nek5000/nekRS#414] [Q&A] [answered]
Hello All,

I have a couple of questions. I need to filter a derivative computed in .udf file because it has some discontinuities.

1) In .usr file I can do something like:

      n = nx1*ny1*nz1*nelv
      call  col2(dwdy, bm1,    n)
      call dssum(dwdy, nx1,    ny1, nz1)
      call  col2(dwdy, bintm1, n)

Is there any way to do this in .udf file?

2) Are there any other filters that I can apply in .udf file? Some sort of filtering or interpolation...

Thank you.

**Comment by stgeke** [comment, marked as answer]
You can average the interface values using:
  ```
    platform->linAlg->axmyVector(
    meshV->Nlocal,
    nrs->fieldOffset,
    0,
    1.0,
    meshV->o_LMM,
    o_fld);
  
  oogs::startFinish(o_fld, /* number of fields */ 1, nrs->fieldOffset, ogsDfloat, ogsAdd, nrs->gsh);

  platform->linAlg->axmyVector(
    meshV->Nlocal,
    nrs->fieldOffset,
    0,
    1.0,
    meshV->o_invLMM,
    o_fld);
 ```
